public/components/event_analytics/explorer/events_views/trace_block/trace_block.tsx

```diff
@@ -43,5 +43,5 @@ export const TraceBlock = ({ http, hit, logTraceId }: props) => {
     );
   }
 
-  return <TraceDetailRender traceId={logTraceId} http={http} mode='data_prepper'/>;
+  return <TraceDetailRender traceId={hit.traceID || logTraceId} http={http} mode={hit.traceID === '' ? 'data_prepper' : 'jaeger'}/>;
```

Comment:
np. would be more clear if you check it at the beginning of the function. reader might not know what's the difference between hit.traceID and logTraceId

```js
const mode = hit.traceID === '' ? 'data_prepper' : 'jaeger';
```

Reply:
Fixed in [f3a871f](https://github.com/opensearch-project/dashboards-observability/pull/150/commits/f3a871f63d3642fa0c25a6de59c1ec6d63e3e3ef)